src/autogluon/bench/eval/evaluation/evaluate_utils.py

```diff
@@ -508,3 +508,26 @@ def compute_win_rate_per_dataset(
     print(f"winrate {f1} vs {f2}")
     print(out_df)
     return out_df
+
+
+def convert_to_dataset_fold(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Converts "{dataset}_{fold}" dataset column to (dataset, fold) columns.
+
+    Parameters
+    ----------
+    df: pandas DataFrame of benchmark results
+        The DataFrame must have a column named `dataset` in the form "{dataset}_{fold}".
+        Example: "adult_5"
+
+    Returns
+    -------
+    Pandas DataFrame of benchmark results with the dataset column split into (dataset, fold) columns
+
+    """
+    df = df.copy(deep=True)
+    df["fold"] = df["dataset"].apply(lambda x: x.rsplit("_", 1)[1])
+    df["dataset"] = df["dataset"].apply(lambda x: x.rsplit("_", 1)[0])
```

Comment:
Is deep copy required?
There is also a built-in function `Series.str.split(expand=True)` for this purpose. So it can be written in
```
df[["dataset", "fold"]] = df["dataset"].str.split("_", expand=True)
```

Reply:
Yes deep copy is required because otherwise you are editing the DataFrame in-place, which would update the outer context.

Re the .str.split, good point! Updated.